### ISSUE TITLE
Lags when I scroll or touch a button

### ISSUE THREAD

**Issue by StendhalSyndrome** [microsoft/vscode#101596]
Version: 1.46.1
Commit: cd9ea6488829f560dc949a8b2fb789f3cdc05f5d
Date: 2020-06-19T10:53:55.306Z
Electron: 7.3.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 5.4.0-39-generic (Ubuntu LTS 20.04)

Steps to Reproduce:

1. Open Visual Studio Code
2. Scroll/touch a button etc...

It's like very low FPS, it's jerky. I already used "window.smoothScrollingWorkaround": true.
When i minimize the window, everything go back to normal (still lags a litlle bit). It does the same thing on Atom, but not on Discord, so it's not a problem common to all electron-app. I'm using an external mouse.

Does this issue occur when all extensions are disabled?: Yes

**Comment by galich** [comment]
seems to be related to https://github.com/microsoft/vscode/issues/97151 

`code --disable-gpu` helped me, but it's not the right answer to the problem

**Comment by deepak1556** [comment]
@StendhalSyndrome can you provide the performance profile with the following steps,

* Command Palette > Toggle Developer Tools
* Go to the `Performance` tab
* Click record profile
* Perform some scrolling that shows the lag
* Stop profiling
* Save the profile

**Comment by deepak1556** [comment]
Also what does the fps meter from devtools indicate when you scroll, you can open the fps meter by 

`Developer Tools > Options > More Tools > Rendering > Fps meter`

**Comment by galich** [comment]
@deepak1556 here is mine from Windows

[Profile-20200715T142209.zip](https://github.com/microsoft/vscode/files/4927232/Profile-20200715T142209.zip)



![image](https://user-images.githubusercontent.com/3718155/87582154-fb872e80-c6a7-11ea-9bcf-4709135d80a4.png)

**Comment by deepak1556** [comment]
@galich thanks for the profile, some quick points

* Each mouse input event are more >1s and the time it waits on the main thread is not even half as close to the total time, indicating something else from script execution is blocking stuff

<img width="1379" alt="Screen Shot 2020-07-15 at 2 41 28 PM" src="https://user-images.githubusercontent.com/964386/87601576-35f1cb00-c6aa-11ea-9a6d-4bcfebf4d933.png">


* Checking all the long tasks (tasks highlighted with red label) reveal that majority of the time is spent is composition layers, which indicates issue with GPU

Can you share your gpu info, also are you using an external mouse ?

The next step would be to perform an internal trace of chromium to find out what exactly is happening at the GPU layer, currently vscode only supports startup chromium tracing, will need to update it before proceeding further. Can you open a new issue to continue the conversation. The authors' issue is on linux which might have different cause, don't want to create noice here.

**Comment by galich** [comment]
@deepak1556 i've filed a new one as requested

https://github.com/microsoft/vscode/issues/102661

**Comment by deepak1556** [comment]
thanks @galich

**Comment by StendhalSyndrome** [comment]
Mine: 
[Profile-20200716T165443.zip](https://github.com/microsoft/vscode/files/4932262/Profile-20200716T165443.zip)

**Comment by StendhalSyndrome** [comment]
![image](https://user-images.githubusercontent.com/41346200/87687587-6eaaa680-c775-11ea-8cda-11be11020811.png)

**Comment by StendhalSyndrome** [comment]
![image](https://user-images.githubusercontent.com/41346200/87687750-a0237200-c775-11ea-9f62-54f7f6f1dbad.png)

**Comment by StendhalSyndrome** [comment]
Looks like the problem has been solved a way or another in the 1.48.0
